includes/Traits/Version_Utils.php

```diff
@@ -0,0 +1,174 @@
+		}
+
+		return $version;
+	}
+
+	/**
+	 * Returns WordPress latest version.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @return string WordPress latest version.
+	 */
+	protected function get_wordpress_latest_version(): string {
+		$version = $this->get_latest_version_info( 'current' );
+
+		return $version ?? get_bloginfo( 'version' );
+	}
+
+	/**
+	 * Returns required PHP version.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @return string Required PHP version.
+	 */
+	protected function get_php_required_version(): string {
+		$version = $this->get_latest_version_info( 'php_version' );
+
+		return $version ?? $this->get_required_php_version();
+	}
+
+	/**
+	 * Returns required MySQL version.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @return string Required MySQL version.
+	 */
+	protected function get_mysql_required_version(): string {
+		$version = $this->get_latest_version_info( 'mysql_version' );
+
+		return $version ?? $this->get_required_mysql_version();
+	}
+
+	/**
+	 * Returns recommended PHP version.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @return string Recommended PHP version.
+	 */
+	protected function get_php_recommended_version(): string {
+		$recommended_version = '7.4'; // Default fallback recommended version.
+
+		$version_details = wp_check_php_version();
+
+		if ( is_array( $version_details ) && ! empty( $version_details['recommended_version'] ) ) {
+			$recommended_version = $version_details['recommended_version'];
+		}
+
+		return $recommended_version;
+	}
+
+	/**
+	 * Returns relative WordPress major version.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @param string $version WordPress major version.
+	 * @param int    $steps   Steps to find relative version. Defaults to 1 for next major version.
+	 * @return string Relative WordPress major version.
+	 */
+	protected function get_wordpress_relative_major_version( string $version, int $steps = 1 ): string {
+		if ( 0 === $steps ) {
+			return $version;
+		}
+
+		$new_version = floatval( $version ) + ( 0.1 * $steps );
+
+		return (string) number_format( $new_version, 1 );
+	}
+
+	/**
+	 * Returns specific information.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @param string $key The information key to retrieve.
+	 * @return mixed The requested information.
+	 */
+	private function get_latest_version_info( string $key ) {
+		$info = get_transient( 'wp_plugin_check_latest_version_info' );
+
+		if ( false === $info ) {
+			$response = wp_remote_get( 'https://api.wordpress.org/core/version-check/1.7/' );
+
+			if ( ! is_wp_error( $response ) && 200 === wp_remote_retrieve_response_code( $response ) ) {
+				$body = json_decode( wp_remote_retrieve_body( $response ), true );
+
+				if ( isset( $body['offers'] ) && ! empty( $body['offers'] ) ) {
+					$info = reset( $body['offers'] );
+					set_transient( 'wp_plugin_check_latest_version_info', $info, DAY_IN_SECONDS );
+				}
+			}
+		}
+
+		return array_key_exists( $key, $info ) ? $info[ $key ] : null;
+	}
+
+	/**
+	 * Returns the current WordPress' required PHP version.
```

Comment:
Why do we need that? Same for the current site's required MySQL version

Reply:
In our internal addon we are trying to use this like checking this value vs PHP version mentioned in the plugin header. We can remove get_mysql_required_version() as I dont see its implementation soon in the near future.

Reply:
We could also use PHP version values in message `The "Requires PHP" header in the plugin file should only contain a PHP version such as "5.2.4" or "7.0".` where currently those versions are hardcoded.

Reply:
Those being hardcoded seems fine to me